core/rlwe/params.go

```diff
@@ -181,10 +183,11 @@ func NewParametersFromLiteral(paramDef ParametersLiteral) (params Parameters, er
 		paramDef.DefaultScale = s
 	}
 
+	// only allow either (Q or P) or (logQ or logP), do not mix log with non-log
```

Comment:
There are use cases where one needs to specify `Q` to ensure that different parameter sets share the same ciphertext modulus (for example switching the ring degree), without having to give specific primes for `P` (i.e. let the library sample them). This prevents such use cases.

Reply:
It's fixed.